app/src/main/java/org/xbmc/kore/utils/Task.java

```diff
@@ -0,0 +1,81 @@
+package org.xbmc.kore.utils;
+
+import android.support.annotation.NonNull;
+
+/**
+ * Represents a computation that will eventually complete.
+ *
+ * @param <T> The type of the value being computed or awaited.
+ */
+public interface Task<T> {
+
+    /**
+     * Starts the computation, invokes a procedure when done.
+     *
+     * @param then The consumer of the value being computed.
```

Comment:
I only implemented the kind of bind which drops the result, but in the last step the result does get passed. It's also the only place where you write a consumer (`OnFinish#got`) in a sequence, so it still makes sense.